### PR TITLE
heartbeat service: specify package

### PR DESCRIPTION
**Pull request by dfithian**
###### Description of changes

Other elastic services (kibana, elasticsearch, filebeat, and metricbeat) allow for a `package` configuration option. This PR adds the same option to heartbeat.

I tested on x86_64 Linux by spinning up a heartbeat service. It ran successfully.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dfithian** [comment]
Hi @fadenb, I noticed you were the one who originally authored this module. I was wondering if you had any thoughts about this change?

**Comment by fadenb** [comment]
Hey @dfithian, not sure whether I should have anything to say as I have not used the module in years 🙈 
From a quick glance the PR seems sensible and I'd say 👍

**Review by fadenb — APPROVED** [review]

**Comment by dfithian** [comment]
@joachifm you merged the original PR. I'm wondering if there's anything I need to change to get this in?

https://github.com/NixOS/nixpkgs/pull/27019

**Comment by joachifm** [comment]
@ddfisher LGTM, if you find this useful.  The only thing I'd change here is make the commit subject conform to the [guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes)

Note that perfectly good PRs can get "stuck" for a long time simply due to "getting lost" in the volume of incoming changes.  If you're a user of this module (or any other module/package, for that matter) please consider requesting maintainer access.

**Pull request merged**
